datafusion/functions/benches/to_timestamp.rs

```diff
@@ -17,97 +17,108 @@
 
 extern crate criterion;
 
+use std::sync::Arc;
+
+use arrow_array::builder::StringBuilder;
+use arrow_array::ArrayRef;
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
 
-use datafusion_expr::lit;
-use datafusion_functions::expr_fn::to_timestamp;
+use datafusion_expr::ColumnarValue;
+use datafusion_functions::datetime::to_timestamp;
 
 fn criterion_benchmark(c: &mut Criterion) {
     c.bench_function("to_timestamp_no_formats", |b| {
-        let inputs = vec![
-            lit("1997-01-31T09:26:56.123Z"),
-            lit("1997-01-31T09:26:56.123-05:00"),
-            lit("1997-01-31 09:26:56.123-05:00"),
-            lit("2023-01-01 04:05:06.789 -08"),
-            lit("1997-01-31T09:26:56.123"),
-            lit("1997-01-31 09:26:56.123"),
-            lit("1997-01-31 09:26:56"),
-            lit("1997-01-31 13:26:56"),
-            lit("1997-01-31 13:26:56+04:00"),
-            lit("1997-01-31"),
-        ];
+        let mut inputs = StringBuilder::new();
```

Comment:
For anyone else following along the previous benchmark was testing how fast we could construct `Expr`s not how fast it actually took to evaluate the function